jboss-eap-config-jgroups/added/launch/ha.sh

```diff
@@ -0,0 +1,120 @@
+        pods_url="https://${KUBERNETES_SERVICE_HOST:-kubernetes.default.svc}:${KUBERNETES_SERVICE_PORT:-443}/api/${OPENSHIFT_KUBE_PING_API_VERSION:-v1}/namespaces/${OPENSHIFT_KUBE_PING_NAMESPACE}/pods"
+        if [ -n "${OPENSHIFT_KUBE_PING_LABELS}" ]; then
+            pods_labels="labels=${OPENSHIFT_KUBE_PING_LABELS}"
+        else
+            pods_labels=""
+        fi
+
+        # make sure the cert exists otherwise use insecure connection
+        if [ -f "${CA_CERT}" ]; then
+            CURL_CERT_OPTION="--cacert ${CA_CERT}"
+        else
+            CURL_CERT_OPTION="-k"
+        fi
+        pods_auth="Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)"
+        pods_code=$(curl --noproxy "*" -s -o /dev/null -w "%{http_code}" -G --data-urlencode "${pods_labels}" ${CURL_CERT_OPTION} -H "${pods_auth}" ${pods_url})
+        if [ "${pods_code}" = "200" ]; then
+            log_info "Service account has sufficient permissions to view pods in kubernetes (HTTP ${pods_code}). Clustering will be available."
+        elif [ "${pods_code}" = "403" ]; then
+            log_warning "Service account has insufficient permissions to view pods in kubernetes (HTTP ${pods_code}). Clustering might be unavailable. Please refer to the documentation for configuration."
+        else
+            log_warning "Service account unable to test permissions to view pods in kubernetes (HTTP ${pods_code}). Clustering might be unavailable. Please refer to the documentation for configuration."
+        fi
+    else
+        log_warning "Environment variable OPENSHIFT_KUBE_PING_NAMESPACE undefined. Clustering will be unavailable. Please refer to the documentation for configuration."
+    fi
+}
+
+validate_dns_ping_settings() {
+  if [ "x$OPENSHIFT_DNS_PING_SERVICE_NAME" = "x" ]; then
+    log_warning "Environment variable OPENSHIFT_DNS_PING_SERVICE_NAME undefined. Clustering will be unavailable. Please refer to the documentation for configuration."
+  fi
+}
+
+validate_ping_protocol() {
+  declare protocol="$1"
+  if [ "${protocol}" = "openshift.KUBE_PING" ]; then
+    check_view_pods_permission
+  elif [ "${protocol}" = "openshift.DNS_PING" ]; then
+    validate_dns_ping_settings
+  else
+    log_warning "Unknown protocol specified for JGroups discovery protocol: $1.  Expecting one of: openshift.KUBE_PING or openshift.DNS_PING."
+  fi
+}
+
+get_socket_binding_for_ping() {
+    # KUBE_PING and DNS_PING don't need socket bindings, but if the protocol is something else, we should allow it
+    declare protocol="$1" socket_binding="$2"
+    local default_socket_binding="socket-binding=\"jgroups-mping\""
+    if [ "${protocol}" = "openshift.KUBE_PING" -o \
+          "${protocol}" = "openshift.DNS_PING" -o \
+          "${protocol}" = "kubernetes.KUBE_PING" -o \
+          "${protocol}" = "dns.DNS_PING" ]; then
+        echo ""
+    else
+        echo "socket-binding=\"${socket_binding_name:-$default_socket_binding}\""
```

Comment:
Ok sure, I like not having the sb param at all. I'll leave the socket binding in the config though, just in case someone has a use for that we don't know about.